pkg/deploy/api/types.go

```diff
@@ -441,6 +441,20 @@ type DeploymentConfigRollbackSpec struct {
 	IncludeStrategy bool
 }
 
+// DeploymentRequest is a request to a deployment config for a new deployment.
+type DeploymentRequest struct {
+	unversioned.TypeMeta
+	// Name of the deployment config for requesting a new deployment.
+	Name string
+	// DryRun if true should not persist the updated deployment config in storage.
+	DryRun bool
+	// Latest will update the deployment config with the latest images found in the cluster.
+	Latest bool
```

Comment:
This is probably not the right scope.  What if we add another trigger type?  I would recommend we identify which triggers should be expected to fire.  If we ever added secrets or config map references as triggers, how would we expand this API to add them.